### PR TITLE
Yet more flinitfy (and related tweaks)

### PR DESCRIPTION
**Pull request by fingolfin**
Also remove redundant `jacobi_symbol` method.

### CODE REVIEW

File: src/antic/nf_elem.jl [hunk 1]
```diff
@@ -396,11 +396,11 @@ function -(a::QQFieldElem, b::AbsSimpleNumFieldElem)
   return r
 end
 
-+(a::AbsSimpleNumFieldElem, b::Integer) = a + ZZRingElem(b)
++(a::AbsSimpleNumFieldElem, b::Integer) = a + flintify(b)
```
Comment (fingolfin) [inline src/antic/nf_elem.jl:399]: test coverage for adhoc operations is really bad. Would be good to have more of those in the interface tests (possibly optionally, as I am not sure we mandate this in the ring interface?)

**Review by lgoettgens — APPROVED** [review]
the matrix constructor changes are very hard to follow. maybe the next time put moving and changing into two different commit? Anyway, I don't see anything bad standing out

File: src/antic/nf_elem.jl (hunk 1 above)
Comment (lgoettgens) [reply src/antic/nf_elem.jl:399]: the ring interface states that it is optional to add explicit ad-hoc functions, but AA should take care of the fallback using promotion (see https://nemocas.github.io/AbstractAlgebra.jl/dev/ring_interface/#Optional-binary-ad-hoc-operators)
